Create currentCard.lcdoc

PaulMcClernan opened the pull request:
I hope this, my first addition here, is correct.
See discussion with Ali regarding current-card in the glossary section.

PaulMcClernan committed:
Create currentCard.lcdoc

I hope my this, my first addition here, is correct.
See discussion with Ali regarding current-card in the glossary section.

PaulMcClernan committed:
Update currentCard.lcdoc

Fixed wouldn't compile with Ali's Docs Helper test stack.

PaulMcClernan committed:
Update currentCard.lcdoc

PaulMcClernan committed:
Update currentCard.lcdoc

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,36 @@
+Syntax: the currentCard
+
+Summary:
+A stack property that is set to the short name of the card in a given stack that is selected for display.
```
This should probably just be `The short name of the <current card> of a stack`, linking to the current card glossary entry.

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,36 @@
+
+Type: property
+
+Syntax: the currentCard
```
Syntax in property entries takes the form

```
Syntax:
get the currentCard of <stack>
set the currentCard of <stack> to <cardName>
```

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,36 @@
+
+-- can be retrieved or set for a stack file that is not currently opened (provided the stack file is writeable)
+
+set currentCard of stack "/Users/admin/Desktop/example.livecode" to "ExampleOfCardNameThatsNotTheFirstCard"
```
The lines in this example should not be double-spaced.

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,36 @@
+Description:
+<currentCard> <property> is a stack property that is set to the short name of the card that is currently selected
+
+for display in a given stack, not neccessarly the front most stack nor necessarily a visible stack or even open stack.
```
- Lines here shouldn't be double-spaced unless they are intended to be separate paragraphs.
- A standard form for Description elements is something like

```
Description:
Use the <currentCard> property to change the <current card> of a stack, or find out which card is the <current card> of a stack.
```

Then maybe you could add something about how the target stack can be not necessarily the frontmost, in contrast to <this> card, perhaps?

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,36 @@
+
+References: current card (glossary)
+
+Tags: card, stack, property, this, current
```
These should be references, they are not tags (and you don't need `current` as it is a synonym of `this`)

I'm not sure if there is a tag that currently applies - maybe `navigation`?

livecodeali commented:
In addition to the comments I've made inline there, there are a couple of other issues:
1. Properties need a `Value:` element that describes the value of the property - this is a good place to say that the value of the currentCard property is the short name, rather than a general card chunk.
2. Properties of objects should have an `Associations:` element with the type of object to which the property pertains (in this case `stack`). This is so that the entry for stack can list all stack properties in a table.

montegoulding commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,36 @@
+Platforms: desktop, server, mobile
+
+Example:
+ -- the following results in the same value as put the short name of the current card of this stack
```
There is an extra space at the start of this line

montegoulding commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,36 @@
+ 
+put the currentCard of this stack
+
+ -- even if the stack is invisible, in the background, or offscreen
```
Here too

peter-b commented:
This needs a release note because it fixes [bug 14651](http://quality.livecode.com/show_bug.cgi?id=14651).

PaulMcClernan committed:
Corrected for proper format 

Having both an Example of an unopened stack file and also including that in the description might be redundant but in might help clarity-wise.

PaulMcClernan committed:
Took out extra leading space

Took out extra leading space before comments per Monte

PaulMcClernan committed:
Added Introduced version

Added Introduced version per bug 14651
http://quality.livecode.com/show_bug.cgi?id=14651

PaulMcClernan committed:
Added property Value: and Associations:

peter-b commented:
Please squash this series on merge.

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,34 @@
+get the currentCard of <stack>
+set the currentCard of <stack> to <cardName>
+
+Value: short name of a card in a stack
```
Can you put a `The` at the beginning of this sentence please? And if possible move the `Value` section to between `Example` and `Description`.

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,34 @@
+set currentCard of stack "/Users/admin/Desktop/example.livecode" to "ExampleOfCardNamedFoo"
+
+Description:
+Use the <currentCard> property to change the <current card> of a stack, or find out which card is the <current card> of a stack. The taget stack does not need to be the frontmost stack, visible and the stack file need need not be open (file write access is needed in order to set the property).
```
This line should be hard-wrapped at 80 characters

PaulMcClernan replied:
I started to make a lcdoc editor stack because of this.
I assume this only applies to the description field as there are lines elsewhere that exceed 80 chars.
Is 80 chars a hard-limit or just a rough number to shoot for?

montegoulding replied:
Most things should be wrapped at 72 characters now. This was scripted a few weeks back. Code wasn't touched because it is harder to hard wrap and syntax doesn't support wrapping. In general it's just links that overrun in other places.

PaulMcClernan replied:
BTW Ali, I DL'd your lcdoc validator stack from dropbox, thanks. It validates but the web preview doesn't seem to work (MacOS X). It opens a browser and then errors. I haven't had a chance to figure out at why.

livecodeali replied:
Which browser?

livecodeali commented on docs/dictionary/property/currentCard.lcdoc:
```diff
@@ -0,0 +1,34 @@
+Description:
+Use the <currentCard> property to change the <current card> of a stack, or find out which card is the <current card> of a stack. The taget stack does not need to be the frontmost stack, visible and the stack file need need not be open (file write access is needed in order to set the property).
+
+References: current card (glossary), card, stack, property, this
```
All the `References` need to have the type specified, so `card (object), stack (object), property (glossary), this (keyword)`

PaulMcClernan committed:
Hard wrap description to 80 chars, others CXs

Hard wrapped description to 80 characters or less.
Tagged References with type
Moved "Value:"...
Misc. text errors fixed

livecodeali commented:
@PaulMcClernan Thanks, almost there! As @peter-b pointed out, this needs a bugfix note added here (https://github.com/PaulMcClernan/livecode/tree/patch-3/docs/notes) as it fixes a bug in Bugzilla. 

There's a section in the contributing to docs document about it here:
https://github.com/livecode/livecode/blob/develop/docs/contributing_to_docs.md#adding-a-bug-fix-release-note

PaulMcClernan committed:
Add release note

PaulMcClernan commented:
Thanks for all of the hand holding guys, I think I got the flow now, cheers!

livecodeali commented:
@livecode-vulcan review ok c036ad6

livecode-vulcan commented:
💙 review by @livecodeali ok c036ad6fa78c670b772426a95697d83ee280181c

livecode-vulcan commented:
😎 test success c036ad6fa78c670b772426a95697d83ee280181c
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/140)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/140)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/140)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/140)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/140)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/140)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/140)
